diff --git a/adnl/adnl-local-id.cpp b/adnl/adnl-local-id.cpp
index ec625352..0d77898c 100644
--- a/adnl/adnl-local-id.cpp
+++ b/adnl/adnl-local-id.cpp
@@ -50,7 +50,7 @@ void AdnlLocalId::receive(td::IPAddress addr, td::BufferSlice data) {
     if (R.is_error()) {
       VLOG(ADNL_NOTICE) << self << ": dropping IN message from " << addr << ": " << R.move_as_error();
     }
-    co_return {};
+    co_return td::Unit{};
   }(this, std::move(addr), std::move(data))
                                                                          .start()
                                                                          .detach();
@@ -92,7 +92,7 @@ td::actor::Task<> AdnlLocalId::receive_coro(td::IPAddress addr, td::BufferSlice
 
   td::actor::send_closure(peer_table_, &AdnlPeerTable::receive_decrypted_packet, short_id_, std::move(packet),
                           data_size);
-  co_return {};
+  co_return td::Unit{};
 }
 
 void AdnlLocalId::deliver(AdnlNodeIdShort src, td::BufferSlice data) {
